tiledb/sm/misc/test/unit_mgc_dict.cc

```diff
@@ -56,42 +55,25 @@
 
 using tiledb::sm::magic_dict;
 
-int check_embedded_data_validity() {
+TEST_CASE("Test embedded data validity", "[mgc_dict][embedded_validity]") {
   FILE* infile = nullptr;
   infile = fopen(TILEDB_PATH_TO_MAGIC_MGC, "rb");
-  if (!infile) {
-    fprintf(stderr, "ERROR: Unable to open %s\n", TILEDB_PATH_TO_MAGIC_MGC);
-    return 1;
-  }
+  REQUIRE(infile);
```

Comment:
I would really prefer that these tests be rewritten with C++ I/O rather than C I/O, but I also admit that it might be a small scope expansion. There's not much I/O, though, so it's not a significant expansion. There's already a need in this PR to do more to update these tests, in addition, so this can go along with that.

Reply:
Rewritten.